clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -573,6 +573,10 @@ public void fireCompletion() {
                 future.raise(response.versionMismatch());
             } else {
                 future.complete(response);
+
+                // dec ref count and release the buffer to memory pool if any
+                // the fetch request buffer pool won't be release since we hold on to buffer references and add ref counts
+                this.response.decRefCount();
```

Comment:
Don't we want to decRefCount in the other else if branches too?

Reply:
Okay, after checking my notes. I think I will make it a little safer.